src/odemis/driver/test/keysight_test.py

```diff
@@ -80,6 +80,20 @@ def test_duty_cycle(self):
         with self.assertRaises(IndexError):
             self.dev.dutyCycle.value = 0.15
 
+        # Change the duty cycle while the power is off: it shouldn't have effect immediately, but
+        # it should be applied when the power is turned on.
+        prev_dev_duty_cycle = self.dev.getDutyCycle(1)  # low-level value (not the same unit)
+        self.dev.power.value = False
+        self.dev.dutyCycle.value = 0.5
+        self.assertEqual(self.dev.dutyCycle.value, 0.5)
+        dev_duty_cycle = self.dev.getDutyCycle(1)
+        self.assertEqual(prev_dev_duty_cycle, dev_duty_cycle)
```

Comment:
I get how this works, but from variable names alone, this behavior is not trivial.

Reply:
Variables names are a good way to express behaviour, better than comments :-)